Added CPack integration for deb and rpm.

cheasles opened the pull request:
<!--- (**********************************)
      (** Fill in the following fields **)
      (**********************************) --->

**Changes proposed:**

-  Enabled CPack integration

**Target branch(es):** 3.3.5/master

- [*] 3.3.5
- [x] master


**Tests performed:**

Ran the following to build a package:

```
mkdir build
cd build
cmake -DCPACK_GENERATOR=DEB  -DCPACK_PACKAGE_CONTACT=cheasles@github.com ..
cmake --build . --target package
```

**Issues**

The most recent full database release is hard-coded into package.cmake.


<!--- Notes
- Enable the setting "[√] Allow edits from maintainers." when creating your pull request.
- If this PR only contains SQL files, open a new issue instead and post or link the SQL in the issue.
- When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.
--->

Commit:
Added CPack integration for deb and rpm.

jackpoz commented:
it seems the CI checks have found some errors:
>CMake Error at cmake/genrev.cmake:68 (list):
  list index: 1 out of range (-1, 0)
Call Stack (most recent call first):
  CMakeLists.txt:83 (include)

>CMake Error at cmake/genrev.cmake:69 (list):
  list index: 2 out of range (-1, 0)
Call Stack (most recent call first):
  CMakeLists.txt:83 (include)

could you describe how is the version of the package retrieved ? like 1.0.0 etc

Shauren commented:
Looks like a cut down version of #24347

jackpoz commented on cmake/package.cmake:
```diff
@@ -0,0 +1,74 @@
+install(DIRECTORY "${CMAKE_SOURCE_DIR}/sql" DESTINATION "${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}")
+
+# Download the full database
+set(FULL_DATABASE_URL "https://github.com/TrinityCore/TrinityCore/releases/download/TDB335.21041/TDB_full_world_335.21041_2021_04_16.7z"
```
This should be updated in https://github.com/jackpoz/TrinityCoreCron/tree/3.3.5-TDB

cheasles replied:
I've made a PR here: https://github.com/jackpoz/TrinityCoreCron/pull/5

Commit:
Version info no longer comes from git and instead hard-coded into package.cmake.

cheasles commented:
It was originally using `git describe --tags` + regex to grab the version information, but I've now changed it to be hard-coded and updated via TrinityCoreCron.

illfated commented:
Nice update. I like the addition of `${DATABASE_TAG}/${DATABASE_NAME}`.

jackpoz commented:
https://ci.appveyor.com/project/DDuarte/trinitycore/builds/39061776 is the only CI left to fix:
>CMake Error at cmake/package.cmake:38 (install):
  install FILES given no DESTINATION!
Call Stack (most recent call first):
  CMakeLists.txt:89 (include)
 Package filename       : trinity-HEAD-318e15895acc-x86_64

Commit:
Fixed for packaging on Windows.

cheasles commented:
I noticed that these changes cause CMake to always download the full world SQL file at generation time, so once CI is passing I'll push a quick cmake option change to enable packaging which is off by default.

Commit:
Disabled package target by default.

cheasles commented:
Looks like everything is working now. Tested locally with the following:

```
mkdir build
cd build
cmake ..
cmake --build . --target package
```

Which failed as expected, then:

```
cmake .. -DBUILD_PACKAGE=1 -DCPACK_GENERATOR=DEB -DCPACK_PACKAGE_CONTACT=cheasles@github.com
cmake --build . --target package
```

Which worked.
